Build with CMake 4, which no longer supports policies older than 3.5.

Drop the request for the OLD behaviour of CMP0022 and stop reading the
LOCATION property of the static libraries (CMP0026) when writing the
libtool control files for the Xvnc build; the library name is
lib<target>.a in the target's binary directory, as TigerVNC 1.9.0 does
it.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f45fbd3..253a6c7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,9 +3,6 @@
 #
 
 cmake_minimum_required(VERSION 2.8)
-if(POLICY CMP0022)
-  cmake_policy(SET CMP0022 OLD)
-endif()
 
 # Internal cmake modules
 set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/Modules)
diff --git a/cmake/Modules/CMakeMacroLibtoolFile.cmake b/cmake/Modules/CMakeMacroLibtoolFile.cmake
index c882c4a..1a3ab7c 100644
--- a/cmake/Modules/CMakeMacroLibtoolFile.cmake
+++ b/cmake/Modules/CMakeMacroLibtoolFile.cmake
@@ -1,5 +1,4 @@
 macro(libtool_create_control_file _target)
-  get_target_property(_target_location ${_target} LOCATION)
   get_target_property(_target_type ${_target} TYPE)
 
   message("-- Creating static libtool control file for target ${_target}")
@@ -113,7 +112,7 @@ macro(libtool_create_control_file _target)
   endforeach()
 
   # Write the libtool control file for the static library
-  get_filename_component(_lname ${_target_location} NAME_WE)
+  set(_lname ${CMAKE_STATIC_LIBRARY_PREFIX}${_target})
   set(_laname ${CMAKE_CURRENT_BINARY_DIR}/${_lname}.la)
 
   file(WRITE ${_laname} "# ${_lname}.la - a libtool library file\n# Generated by ltmain.sh (GNU libtool) 2.2.6b\n")
@@ -142,6 +141,6 @@ macro(libtool_create_control_file _target)
   add_custom_command(TARGET ${_target} POST_BUILD COMMAND 
     "${CMAKE_COMMAND}" -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/.libs")
   add_custom_command(TARGET ${_target} POST_BUILD COMMAND
-    "${CMAKE_COMMAND}" -E create_symlink ${_target_location} "${CMAKE_CURRENT_BINARY_DIR}/.libs/${_lname}${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    "${CMAKE_COMMAND}" -E create_symlink ../${_lname}${CMAKE_STATIC_LIBRARY_SUFFIX} "${CMAKE_CURRENT_BINARY_DIR}/.libs/${_lname}${CMAKE_STATIC_LIBRARY_SUFFIX}")
 
 endmacro()
